### PR TITLE
refactor(sdk): bigint internals, fix duplicate channels, consolidate tests

### PR DESCRIPTION
**Pull request by ittaysw**
- Internal code uses bigint exclusively; external interfaces normalize BigNumberish
- Consolidate tests to integration.test.ts and edge-cases.test.ts (10 tests)
- Add .claude to .gitignore

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/starknet-privacy/198)

### CODE REVIEW

**Comment by ittaysw** [comment]
* **#198** <a href="https://app.graphite.com/github/pr/starkware-libs/starknet-privacy/198?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/starkware-libs/starknet-privacy/198?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#187** <a href="https://app.graphite.com/github/pr/starkware-libs/starknet-privacy/187?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#192](https://github.com/starkware-libs/starknet-privacy/pull/192) <a href="https://app.graphite.com/github/pr/starkware-libs/starknet-privacy/192?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/starkware-libs?ref=merge-queue-instructions-in-stack-comment&prId=198).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: sdk/src/internal/builders.ts [hunk 1]
```diff
@@ -36,73 +42,90 @@ export class TokenOperationsBuilderImpl implements TokenOperationsBuilder {
...
-      if ("noteId" in input) {
-        this.deposits.push({ token: this.token, amount: input.amount, noteId: input.noteId });
-      } else {
-        this.deposits.push({ token: this.token, amount: input.amount });
-        this.createNotes.push({
-          token: this.token,
-          recipient: input.recipient ?? this.parentBuilder.userAddress,
-          amount: input.amount,
-        });
+      // if ("noteId" in input) {
```
Comment (m-kus) [inline sdk/src/internal/builders.ts]: Can we remove the commented out code?

Comment (ittaysw) [reply sdk/src/internal/builders.ts]: done

**Review by m-kus — APPROVED** [review]
